sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2538,6 +2538,56 @@ case class Encode(value: Expression, charset: Expression)
     newLeft: Expression, newRight: Expression): Encode = copy(value = newLeft, charset = newRight)
 }
 
+/**
+ * Converts the input expression to a binary value based on the supplied format.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str[, fmt]) - Converts the input `str` to a binary value based on the supplied `fmt`.
+      By default, the binary format for conversion is "hex" if `fmt` is omitted.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('abc', 'utf-8');
+       abc
+  """,
+  since = "3.3.0",
+  group = "string_funcs")
+// scalastyle:on line.size.limit
+case class ToBinary(expr: Expression, format: Option[Expression], child: Expression)
+  extends RuntimeReplaceable {
+
+  def this(expr: Expression, format: Expression) = this(expr, Option(format),
+    format.toString match {
+      case "hex" => Unhex(expr)
```

Comment:
We should probably check
1. if lower cases of them are supported or not.
2. if foldable expressions are allowed for `format` (e.g., `to_binary(..., 'base' + '64')`) in other references of DBMSes

Reply:
`format` is insensitive in most of DBRMSes, such as SNOWSQL, Amazon Redshift, and BigQuery.

Reply:
[7490be7](https://github.com/apache/spark/pull/35415/commits/7490be7f97a9a9fda223eabec95fa481f6f89984) supports cases as below
```
scala> spark.sql("select to_binary('abc', 'HEx')").show()
+-------------------+
|to_binary(abc, HEx)|
+-------------------+
|            [BC 00]|
+-------------------+

scala> spark.sql("select to_binary('abc', 'uTF-8
+---------------------+
|to_binary(abc, uTF-8)|
+---------------------+
|           [61 62 63]|
+---------------------+
```
However, it doesn't function fully as we may see from `src/test/resources/sql-tests/results/string-functions.sql.out`. 

I suspect it to be related to `UTFString`. Do you have any idea @HyukjinKwon , thanks!

Reply:
With [f49f7cf](https://github.com/apache/spark/pull/35415/commits/f49f7cf48a79d1f75cf6ee920c88a50b4f0eee12), we support cases as below.

```
scala> spark.sql("select to_binary('abc', 'HEx')").show()
+-------------------+
|to_binary(abc, HEx)|
+-------------------+
|            [BC 00]|
+-------------------+

scala> spark.sql("select to_binary('abc', 'uTF-8')").show()
+---------------------+
|to_binary(abc, uTF-8)|
+---------------------+
|           [61 62 63]|
+---------------------+
```